R/water_price.R

```diff
@@ -19,34 +20,44 @@
 #'   }
 #' 
 
-water_price <- function(gdx, file=NULL, level="reg", index=FALSE, index_baseyear=2005, digits=4) {
+water_price <- function(gdx, file=NULL, level="reg", weight = "value", index=FALSE, index_baseyear=2005, digits=4) {
+  
+  #cellular level
   oq_water_cell <- readGDX(gdx,"oq43_water","oq_water", format="first_found")[,,"marginal"]
-  oq_water_cell<- as.magpie(round(oq_water_cell,digits))
+  oq_water_cell <- as.magpie(round(oq_water_cell,digits=digits))
   
-  if(is.null(oq_water_cell)) {
-    warning("Water shadow prices cannot be calculated as needed data could not be found in GDX file! NULL is returned!")
-    return(NULL)
-  }
-
-  if(level=="cell"){
+  ovm_watdem_cell <- setNames(readGDX(gdx,"ov_watdem","ov43_watdem","ovm_watdem", format="first_found")[,,"agriculture.level"],NULL)
+  
+  #cluster level water consumption
+  weight_cell <- -oq_water_cell*ovm_watdem_cell
```

Comment:
You can define your `weight_cell` values here, based on the argument `weight`. It's less code then to apply different weights down under when you do the calculation.